openedx/core/djangoapps/catalog/utils.py

```diff
@@ -119,6 +119,30 @@ def get_program_types(name=None):
         return []
 
 
+def get_currency():
+    """Retrieve currency data from the catalog service.
+
+    Returns:
+        list of dict, representing program types.
+        dict, if a specific program type is requested.
+    """
+    catalog_integration = CatalogIntegration.current()
+    if catalog_integration.enabled:
+        try:
+            user = catalog_integration.get_service_user()
+        except ObjectDoesNotExist:
+            return []
+
+        api = create_catalog_api_client(user)
+        cache_key = '{base}.currency'.format(base=catalog_integration.CACHE_KEY)
```

Comment:
Nit: move the `if / else` in the `get_edx_api_data` call to here? It just seems we're doing implicit assignment of the `cache_key` value in two places so combine it?

```python
cache_key = '{base}.currency'.format(base=catalog_integration.CACHE_KEY) \ 
     if catalog_integration.is_cache_enabled \
     else None
```

(Excuse the line breaks - I don't know what the best format is exactly)

Reply:
i'd rather just stay consistent with the other methods in this file